cmd/admin-handlers.go

```diff
@@ -1799,3 +1799,66 @@ func (a adminAPIHandlers) TraceHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+// The handler sends console logs to the connected HTTP client.
+func (a adminAPIHandlers) ConsoleLogHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ConsoleLog")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	node := r.URL.Query().Get("node")
+	// limit buffered console entries if client requested it.
+	limitStr := r.URL.Query().Get("limit")
+	limitLines, err := strconv.Atoi(limitStr)
+	if err != nil {
+		limitLines = 10
+	}
+	// Avoid reusing tcp connection if read timeout is hit
+	// This is needed to make r.Context().Done() work as
+	// expected in case of read timeout
+	w.Header().Add("Connection", "close")
+	w.Header().Set(xhttp.ContentType, "text/event-stream")
+
+	doneCh := make(chan struct{})
+	defer close(doneCh)
+	logCh := make(chan interface{}, 4000)
+
+	remoteHosts := getRemoteHosts(globalEndpoints)
+	peers, err := getRestClients(remoteHosts)
+	if err != nil {
+		return
+	}
+
+	globalConsoleSys.Subscribe(logCh, doneCh, node, limitLines, nil)
+
+	for _, peer := range peers {
+		peer.ConsoleLog(logCh, doneCh)
```

Comment:
Shouldn't we listen for the peer logs only if `node != thishost.Name`. As far as i understand, we need peer logs only if  `node == ""` (or) `node == one among the remote peer nodes`. If So, in this case, we could ask the peer logs only if the node is not set (or)  the node is set to any of the remote peer's hostname.

Reply:
AFAICS two clients requesting logs are just two different queries right - ie, two separate queries are served by two minio processes with respective channels. Or dont they work independently of each other?. For example, if one query has a `local node` in node parameter. that request need not serve logs from other peers. So we can simply ignore peer logs. And if another query comes with a remote peer node in node parameter, then we could register/subscribe that single peer alone / all peers and ignoring the local logs.

Reply:
yes, that's a good catch - fixed this.